File: Excel/449 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/449 Rotated Strings.xlsx", range = "A1:B10") %>% arrange(String1)
test  = read_excel("Excel/449 Rotated Strings.xlsx", range = "C1:D6") %>% arrange(`Answer Expected`)
colnames(test) = colnames(input)

# approach 1

is_rotated = function(string1, string2) {
  is_0_rot = string1 == string2
  is_rot = str_detect(paste0(string1, string1), string2)
  return(is_rot & !is_0_rot)
}

result = input %>%
  mutate(is_rotated = map2_lgl(String1, String2, is_rotated)) %>%
  filter(is_rotated) %>%
  select(-is_rotated)

identical(result, test)
# [1] TRUE


# approach 2 

result2 = input %>% 
  filter(map2_lgl(String1, String2, ~str_detect(paste0(.x, .x), .y) & .x != .y))

identical(result2, test)
# [1] TRUE

File: Excel/400-499/449/449 Challenge.py
import pandas as pd

input = pd.read_excel("449 Rotated Strings.xlsx",  usecols="A:B").sort_values(by='String1')
test = pd.read_excel("449 Rotated Strings.xlsx",  usecols="C:D", nrows=5)
test.columns = input.columns
test = test.sort_values(by='String1').reset_index(drop=True)

def is_rotated(original, check):
    return check in original + original and original != check and len(original) == len(check)

input['Rotated'] = input.apply(lambda x: is_rotated(x['String1'], x['String2']), axis=1)
input = input[input['Rotated'] == True].drop(columns='Rotated').reset_index(drop=True)

print(input.equals(test)) # True

input2  = input[input.apply(lambda x: x['String2'] in (x['String1'] + x['String1']) and x['String1'] != x['String2'] and len(x['String1']) == len(x['String2']), axis=1)].reset_index(drop=True)

print(input2.equals(test)) # True